gen/cycle.ml

```diff
@@ -463,16 +463,16 @@ module CoSt = struct
 
   let get_cell st = st.co_cell
 
-  let set_cell ?(rv = -1) st n =
+  let set_cell st n =
     let e = n.evt in match e.bank with
     | Ord -> begin
        let old = st.co_cell.(0) in
        let cell = E.overwrite_value old e.atom e.v in
        let co_cell = Array.copy st.co_cell in
-       let cell2 = match n.edge.E.edge with
+       let cell2 = match n.prev.edge.E.edge with
         | E.Rmw rmw -> E.compute_rmw rmw old cell
-        | _ -> if rv = -1 then cell else rv in
-      co_cell.(0) <- cell2;
+        | _ -> cell in
+       co_cell.(0) <- cell2;
       {e with cell=co_cell;},{ st with co_cell; }
```

Comment:
Performing a RWM operation is as follows:

1. Read old value, here `let old = st.co_cell.(0)`
2. Compute, here `let cell2 = ... E.compute_rmw old e.v`
3. Store the result, here  `co_cell .(0) <- E.overwrite_value old e.atom cell2`

The `E.overwrite` function replaces the the subpart of the memory location specified by `e.atom` with the operation result, leaving the rest unchanged. This applies only to mixed-size mode, where `e.atom` may specify some subpart such as second byte, first 16bits word, etc.

Reply:
Thank you!